modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/BasicIndexTest.java

```diff
@@ -229,6 +257,44 @@ public void testNoIndexesNoPersistence() throws Exception {
         }
     }
 
+    /** */
+    @Test
+    public void testInStatementUsesIndex() throws Exception {
+        H2TreeIndex.h2TreeFactory = TestH2Tree::new;
+
+        inlineSize = 30;
+
+        IgniteEx ig0 = startGrid();
+        GridQueryProcessor qryProc = ig0.context().query();
+
+        Function<String, FieldsQueryCursor<List<?>>> execSql =
+            sql0 -> qryProc.querySqlFields(new SqlFieldsQuery(sql0), true);
+
+        execSql.apply("CREATE TABLE PUBLIC.TEST_TABLE (f1 int, f2 int, f3 int, f4 int, PRIMARY KEY (f1, f2, f3))");
+        execSql.apply("CREATE INDEX PK_IDX ON PUBLIC.TEST_TABLE (f1, f2, f3)");
```

Comment:
why do we need such index ? test pass without it too.

Reply:
Can you expand tests for case like :
`        execSql.apply("CREATE TABLE PUBLIC.TEST_TABLE (f1 int, f2 int, f3 int, f4 int, PRIMARY KEY (f1))");
        execSql.apply("CREATE INDEX PK_IDX ON PUBLIC.TEST_TABLE (f1, f2, f3)");`
        
`        execAssert.accept("SELECT * FROM PUBLIC.TEST_TABLE USE INDEX(PK_IDX) WHERE f1 = 8 and f2 in (1, 5, 3, 7) and f3 = 5");
        execAssert.accept("SELECT * FROM PUBLIC.TEST_TABLE USE INDEX(PK_IDX) WHERE f1 in (1, 5, 3, 7) and f2 = 8 and f3 = 5");
        execAssert.accept("SELECT * FROM PUBLIC.TEST_TABLE USE INDEX(PK_IDX) WHERE f1 = 8 and f2 = 5 and f3 in (1, 5, 3, 7)");`

Reply:
Thanks, I updated the test, but since the PK must be unique, I added a separate column for it.
Please take a look.